core/drivers/stm32_tamp.c

```diff
@@ -361,23 +360,33 @@ static TEE_Result stm32_tamp_apply_bkpr_rif_conf(void)
 	return TEE_SUCCESS;
 }
 
-TEE_Result stm32_tamp_set_secure_bkpregs(struct stm32_bkpregs_conf *bkr_conf)
+/*
+ * stm32_tamp_set_secure_bkprwregs: Configure backup registers zone.
+ * registers in zone 1: read/write only in secure mode
+ *              zone 2: write only in secure mode, read in secure and
+ *                       non-secure mode
+ *              zone 3: read/write in secure and non-secure mode
+ *
+ * return TEE_ERROR_NOT_SUPPORTED:  if zone 1 and/or zone 2 definition are out
+ *                                  of range.
+ *        TEE_ERROR_BAD_PARAMETERS: if bkpregs_cond is NULL.
+ *        TEE_SUCCESS             : if OK.
+ */
+static TEE_Result stm32_tamp_set_secure_bkpregs(void)
 {
-	struct stm32_tamp_instance *tamp = stm32_tamp_dev;
+	struct stm32_bkpregs_conf bkpregs_conf =
+		stm32_tamp_dev->pdata.bkpregs_conf;
```

Comment:
nitpicking: this copies data whereas using a pointer would be enough. No functional change. Fix or keep as-is at your will. Ditto in `stm32_tamp_apply_bkpr_rif_conf()`.